library/pod-security-policy/host-filesystem/template.yaml

```diff
@@ -16,13 +16,16 @@ spec:
           properties:
             allowedHostPaths:
               type: array
+              description: "An array of hostpath objects, representing paths and read/write configuration."
               items:
                 type: object
                 properties:
                   readOnly:
                     type: boolean
+                    description: "when set to true, will allow host volumes matching the pathPrefix only if all volume mounts are readOnly."
```

Comment:
all volume mounts or all matching volume mounts?

Reply:
So, hilariously, I copied this [from the k8s libraries themselves](https://github.com/kubernetes/api/blob/96087243fecba390391364dd42abe2f3ff21c1e3/policy/v1beta1/types.go#L328).

> all volume mounts or all matching volume mounts?

So this would be all of the volume mounts that match that prefix.  So, if you had another allowed prefix with `readOnly: false` the mounts matching _that_ prefix would _not_ be required to be `readOnly`.

---

Another version of that description could be:

`When set to true, all mounts covered by the pathPrefix must be mounted readOnly.`

Thoughts?

Reply:
Ah, def. don't just use PSP words:

1) Because we should at least cite them if we do
2) Because the Rego may not be written exactly like the PSP Go code... particularly if the behavior isn't documented as part of the API

Looking at the template, here's what it seems like is happening:

- It's possible to use paths on the host node as volumes that can be mounted to the pods (`spec.volumes`)
   - in order to do this, you must specify that the volume is a host path type, and which host path you want to use as a volume, this path is what `pathPrefix` matches against
- A volume can be mounted to one or more paths in the pod
  - Each volume mount must reference a volume defined in `spec.volumes`
  - Each volume mount specifies where that volume gets mounted inside the pod (this value is not checked)
  - Each volume mount can be mounted as read only
  
  
It appears that the `readOnly` parameter allows users to say "for every volume mount that references this volume must mount it as read only"

Reply:
how about:

```
when set to true, any container volumeMounts matching the pathPrefix must include `readOnly: true`
```